### PR TITLE
new params for deploy template

### PR DESCRIPTION
**Pull request by vesalink**

### CODE REVIEW

File: lib/templates/openshift/deploy.yaml [hunk 1]
```diff
@@ -270,5 +270,11 @@ parameters:
...
 - name: INDEX
   description: index page for application
-
+- name: OPENJDK18_IMAGE_NAME
+- name: TOMCAT_IMAGE_TAG
+- name: ARTIFACT_VERSION
+- name: ARTIFACT_ID
+- name: PACKAGING_TYPE
+- name: OPENJDK18_IMAGE_TAG
+- name: TOMCAT_IMAGE_NAME
```
Comment (ianliu28) [inline lib/templates/openshift/deploy.yaml]: Hello Sergey,
I am not very sure, but these parameters look like parameters for 'build' template, not for 'deploy' template.
Also I don't see you change code to use them.
Could you check again to make sure you do need these parameteres, or they may become garbage code from time to time if we don't use them.
Thank you, just let me know after you check and the reason why they are there without using them.

Comment (vesalink) [reply lib/templates/openshift/deploy.yaml]: no, these are deploy params as well, see here: https://github.com/BCDevOps/nr-pipeline-ext/blob/release/0.0.1/lib/basicJavaAppDeploy.js#L24 build and deploy params merged, so if they missed in deploy template, deploy will fail... I've tested it already...

Comment (ianliu28) [reply lib/templates/openshift/deploy.yaml]: Hm.... Sergey, thanks for point that out to me. 
I think this '[basicJavaAppDeploy.js]( https://github.com/BCDevOps/nr-pipeline-ext/blob/release/0.0.1/lib/basicJavaAppDeploy.js#L24)' was recently being migrated from Poornima, but I am pretty sure these few params should not be used in 'deploy'. She probably forgot to commit/push her changes to this file.
If you looked at recent working deployment in[ dev from IRS](https://apps.nrs.gov.bc.ca/int/stash/projects/IRS/repos/irs-war/browse/openshift/deploy.yaml?at=refs%2Fheads%2FIRS-149-deploy-irs-to-openshift-dlvr-team-zero-test-task), which she used her script to generated the template for deployment, it does not contain these parameters.
The reason that should not be there are:
1. you can compare previous siwe/wiof deployment templates and you will see why.
2. These parameters make sense only for building images, not during deploy.
3. If you look at new [config.js](https://github.com/BCDevOps/nr-pipeline-ext/blob/release/0.0.1/lib/templates/.pipeline/lib/config.js) she created under nr-pipeline-ext (not the project itself) you will see she probably meant to use "**deployParams**" 
```
const buildParams = {
    OPENJDK18_IMAGE_TAG: '1.6-23',
    OPENJDK18_IMAGE_NAME: 'openjdk18-openshift',
    TOMCAT_IMAGE_TAG: '9.0.30',
    TOMCAT_IMAGE_NAME: 'ha-tomcat9',
    WEBAPP_NAME: '',
    ARTIFACT_ID: '',
    PACKAGING_TYPE: '',
    ARTIFACT_VERSION: '',
}
const deployParams = {}
```
So it is a typo in '**basicJavaAppDeploy.js**' to me. You probably need to fix this in that javascript, not the deploy template.
However, current deployParams={}, I hope it is correct or you may need to find out if you need to add some general params in. Need to test on that though....
Hope it make sense to you or let me know if you have any question.

Comment (vesalink) [reply lib/templates/openshift/deploy.yaml]: yeah, you right, there is some mix with build/deploy variables... I've updated the pr

**Review by ianliu28 — APPROVED** [review]

**Pull request merged**